lib/msal-node/src/client/ClientApplication.ts

```diff
@@ -83,14 +84,21 @@ export abstract class ClientApplication {
      */
     async acquireTokenByCode(request: AuthorizationCodeRequest): Promise<AuthenticationResult> {
         this.logger.info("acquireTokenByCode called");
-        const authClientConfig = await this.buildOauthClientConfiguration(
-            request.authority
-        );
-        this.logger.verbose("Auth client config generated");
-        const authorizationCodeClient = new AuthorizationCodeClient(
-            authClientConfig
-        );
-        return authorizationCodeClient.acquireToken(this.initializeRequestScopes(request) as AuthorizationCodeRequest);
+        const validRequest = this.initializeRequest(request) as AuthorizationCodeRequest;
```

Comment:
Might make more sense to pass the TelemetryManager into the `buildOauthClientConfiguration` then to each acquireTokenCall.

Reply:
Yeah I thought about that. Are we reasonably certain that we will always create a new client for each request? Or could that change in the future? That was the only reason I thought it was better to add it to the acquireTokenCall

Reply:
Yea a new client will be created for each request. (It would probably make more sense to name them "requests" instead of "clients", but that's a separate issue)

Reply:
Ended up taking both of these suggestions. Code is much cleaner now